src/backend/api/meals.js

```diff
@@ -1,15 +1,179 @@
 const express = require("express");
+const Knex = require("knex");
+const { groupBy } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
+  const { maxPrice, availableReservations, title, createdAfter, limit } = unpackParams(request); // This step will throw if any params are invalid
+  const reservation = await knex('reservation')
+
+  let meals = knex('meal')
+
+  if (maxPrice != Number.MAX_VALUE) {
+    meals = meals.where('price', '<', maxPrice)
+  }
+  if(title.length!==0){
+    meals=meals.where('title','LIKE',`%${title}%`)
+  }
+  if(createdAfter) {
+    meals=meals.where('created_date','>',createdAfter)
+  }
+if(limit!==Number.MAX_VALUE) {
+  meals=meals.limit(limit)
+}
+
+  
+
+  if (availableReservations==true) {
+    meals
+      .join("reservation", "meal.id", "reservations.meal_id")
+      .select(
+        "meal.id",
+        "title",
+        "description",
+        "location",
+        "when",
+        "created",
+        "price",
+        "max_reservation",
+        knex.raw("SUM(number_of_guests) AS total")
+      )
+      .where("max_reservation", ">", "number_of_guests")
+      .groupBy("meal_id").having(Knex.raw('sum(max_reservation-SUM(number_of_guests>0'))
```

Comment:
I don't think this will work. there is something wrong with the pairing of parenthesis.

Reply:
Yes thanks ,I have implemented the changes